core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/ProgressiveUtils.scala

```diff
@@ -10,14 +15,14 @@ object ProgressiveUtils {
   // false indicates retraction (-)
   val insertRetractFlagAttr = new Attribute("__internal_is_insertion", AttributeType.BOOLEAN)
 
-  def addInsertionFlag(tuple: Tuple): Tuple = {
+  def addInsertionFlag(tuple: Tuple, outputSchema: Schema): Tuple = {
     assert(!tuple.getSchema.containsAttribute(insertRetractFlagAttr.getName))
-    Tuple.newBuilder.add(insertRetractFlagAttr, true).add(tuple).build
+    Tuple.newBuilder(outputSchema).add(insertRetractFlagAttr, true).add(tuple).build
```

Comment:
@zuozhiw please review this file changes

Reply:
I think this change is correct, as long as the output schema is the new schema, which I think it is the case

Reply:
Yes. This is the output schema that's being passed in as a parameter.